lib/logstash/inputs/sqs.rb

```diff
@@ -73,104 +74,105 @@ class LogStash::Inputs::SQS < LogStash::Inputs::Threadable
+    event[@md5_field] = message.md5_of_body if @md5_field
+    event[@sent_timestamp_field] = convert_epoch_to_timestamp(message.attributes[SENT_TIMESTAMP]) if @sent_timestamp_field
+
+    return event
+  end
+
+  def handle_message(message)
+    event = decode_event(message)
+    add_sqs_data(event, message)
+    decorate(event)
+    return event
+  end
+
+  def run(output_queue)
+    @logger.debug("Polling SQS queue", :polling_options => polling_options)
+
+    run_with_backoff do
+      poller.poll(polling_options) do |messages, stats|
+        messages.each do |message|
+          output_queue << handle_message(message)
+        end
+
+        @logger.debug("SQS Stats:", :request_count => stats.request_count,
+                      :messages_count => stats.message_count,
+                      :last_message_received_at => stats.last_message_received_at) if @logger.debug?
+      end
+    end
+  end
 
   private
   # Runs an AWS request inside a Ruby block with an exponential backoff in case
-  # we exceed the allowed AWS RequestLimit.
+  # we experience a ServiceError.
   #
   # @param [Integer] max_time maximum amount of time to sleep before giving up.
   # @param [Integer] sleep_time the initial amount of time to sleep before retrying.
   # @param [Block] block Ruby code block to execute.
-  def run_with_backoff(max_time, sleep_time, &block)
-    if sleep_time > max_time
-      @logger.error("AWS::EC2::Errors::RequestLimitExceeded ... failed.", :queue => @queue)
-      return false
-    end # retry limit exceeded
+  def run_with_backoff(max_time = MAX_TIME_BEFORE_GIVING_UP, sleep_time = BACKOFF_SLEEP_TIME, &block)
+    next_sleep = sleep_time
 
     begin
       block.call
-    rescue AWS::EC2::Errors::RequestLimitExceeded
-      @logger.info("AWS::EC2::Errors::RequestLimitExceeded ... retrying SQS request", :queue => @queue, :sleep_time => sleep_time)
-      sleep sleep_time
-      run_with_backoff(max_time, sleep_time * 2, &block)
-    rescue AWS::EC2::Errors::InstanceLimitExceeded
-      @logger.warn("AWS::EC2::Errors::InstanceLimitExceeded ... aborting SQS message retreival.")
-      return false
-    rescue AWS::SQS::Errors::InternalError
-      @logger.info("AWS::SQS::Errors::AWS Internal Error ... retrying SQS request with exponential backoff", :queue => @queue, :sleep_time => sleep_time)
-      sleep sleep_time
-      run_with_backoff(max_time, sleep_time * 2, &block)
-    rescue Exception => bang
-      @logger.error("Error reading SQS queue.", :error => bang, :queue => @queue)
-      return false
-    end # begin/rescue
-    return true
-  end # def run_with_backoff
+      next_sleep = sleep_time
+    rescue Aws::SQS::Errors::ServiceError => e
+      @logger.info("Aws::SQS::Errors::ServiceError ... retrying SQS request with exponential backoff", :queue => @queue, :sleep_time => sleep_time, :error => e)
+      sleep(next_sleep)
+      next_sleep =  next_sleep > max_time ? sleep_time : next_sleep = sleep_time * BACKOFF_FACTOR 
```

Comment:
@jsvd Can you check if you agree with this logic? 

In the current version of the plugin we have a max retries of SQS specific errors before silently killing the input thread.
I've changed that so the backoff will keep retrying (using a max sleep time), I mean theses errors should be transient but 60 seconds could happen and the input should not be killed for that.